api/src/main/java/bio/terra/tanagra/service/search/Variable.java

```diff
@@ -0,0 +1,13 @@
+package bio.terra.tanagra.service.search;
+
+import com.google.auto.value.AutoValue;
+
+/** A variable to alias an entity to allow referring to multiple aliases of the same entity. */
+@AutoValue
+public abstract class Variable {
```

Comment:
Thoughts on Alias vs Variable for this class name? Or does this actually encompass both cases of normal references and aliases (if not, what is the value of `name` in the case where there's no alias, or does this model just not exist there)

Reply:
I'm thinking of this in terms of https://en.wikipedia.org/wiki/Tuple_relational_calculus , where this is a tuple variable.
I think "variable" is a better fit. So far, variables always relate to a given entity, but having variables allows us to refer to multiple instances of the same entity in a given query - we need this for temporal queries, e.g. we want to relate 2 Visits x & y by their visit date ( x.visit_date = y.visit_date).
Later, I think variables will also be useful for aggregations, e.g. at least 5 procedures match some condition.
`WHERE (SELECT COUNT(procedure) as procedure_count FROM procedure WHERE ...) x
  AND x.procedure_count > 5`
So here x is binding to some "aggregated entity." but I'm not there yet.

But I'm really struggling to document this in a helpful way. Happy to talk more offline and iterate.

Reply:
This seems fine and probably correct, mostly just probing a bit for my own comprehension. Discussion not blocking, I'll add to agenda for when we next talk